Adds Difference and Differentiator blocks.

Tirpitz93 opened the pull request:
Adds Difference and Differentiator blocks.
I think this would be best placed in the signals library in core once everyone is happy with it.

Tirpitz93 committed:
Adds Difference and Differentiator blocks.

Tirpitz93 commented:
https://github.com/Hopsan/hopsan/issues/2143

robbr48 commented on diff/diffDiff.hpp:
```diff
@@ -0,0 +1,126 @@
+        //Configure
+        void configure()
+        {
+            //Register constants
```
These empty comments can be removed (also on other places)

Tirpitz93 replied:
Removed

robbr48 commented on diff/diffDiff.hpp:
```diff
@@ -0,0 +1,126 @@
+
+            //Write output variables
+            (*mpOut1) = In1 - prev;
+            prev = In1;
```
If we assign `prev` before updating `In1` above we can change it from a member variable to local variable.

Tirpitz93 replied:
good spot,  completed in 4d225cd

robbr48 commented on diff/diffDiff.hpp:
```diff
@@ -0,0 +1,126 @@
+
+            //Add ports
+            addInputVariable("In1", "Input", "", 0, &mpIn1);
+            addOutputVariable("Out1", "Output", "", 0, &mpOut1);
```
Please just call the ports "in" and "out", to be consistent with other signal components.

robbr48 replied:
Looks good to me, just some minor comments. I only commented on diffDiff.hpp, but the same comments apply to both hpp files. If you want to you could also rename the components to "SignalDifference" and "SignalDifferentiator". This would make it easier to move it to the default library later (which I think we should do).

Tirpitz93 replied:
Completed in 4d225cd

Tirpitz93 committed:
Addresses PR Comments

In/out Variable renamed to `in` and `out`.
Empty methods removed
Variable `prev` changed from member to private variable
Blocks renamed to `SignalDifference` & `SignalDifferentiator` respectively.
Removes redundant includes.

robbr48 reviewed (approved):
Looks good to me! I think we can move these components to the default library quite soon, after some testing.
